File: Cliente-Java/Cliente.java
import java.io.*;
import java.net.*;
import java.util.Scanner;

public class Cliente {
    public static void main(String[] args) throws UnknownHostException, IOException {

        System.out.println("CLIENTE JAVA");
        Scanner scanner = new Scanner(System.in);

        String address = "127.0.0.1";
        int port = 5000;
        Socket socket;

        int option = 0;
        int length = 0;


        while (option != 3) {

            option = Functions.printMenu(scanner);

            switch (option) {
                case 1:
                    //INGRESO LONGITUD
                    System.out.println("Ingrese la longitud del nombre de usuario");
                    length = Functions.readLength(scanner);

                    //REALIZAR REQUEST
                    socket = Functions.send(option, length, scanner,address,port);

                    //RECIBIR RESPUESTA
                    Functions.recieve(socket,5,15);

                    //CERRAR CONEXION
                    socket.close();
                    break;
                case 2:
                    //INGRESO LONGITUD
                    System.out.println("Ingrese la longitud del nombre de la contraseña");
                    length = Functions.readLength(scanner);

                    //REALIZAR REQUEST
                    socket = Functions.send(option, length, scanner,address,port);

                    //RECIBIR RESPUESTA
                    Functions.recieve(socket,8,49);

                    //CERRAR CONEXION
                    socket.close();
                    break;
                case 3:
                    //Cerrar aplicación
                    System.out.println("Saliendo...");
                    break;
                default:

                    System.out.println("Debe ingresar un valod del 1 al 3");
                    break;
            }
        }
    }
}

File: Cliente-C/cliente.c
#include<WinSock2.h>
#include<stdio.h>
#include<windows.h>
#include "funciones/funciones.h"

int PUERTO = 5000;

void main(void)
{
    int option = 0;
    int length = 0;

    /* Inicializar WinSock en la aplicación Cliente */
    WSADATA WsaData;
    WSAStartup(MAKEWORD(2, 2), &WsaData);

    /*Crear socket*/
    SOCKET Sock;

    /*Configurar a donde intentará conectarse el cliente*/
    SOCKADDR_IN serverAddress;
    memset (&serverAddress, 0, sizeof(serverAddress));
    serverAddress.sin_family = AF_INET;
    serverAddress.sin_addr.s_addr = inet_addr("127.0.0.1");
    serverAddress.sin_port = htons(PUERTO);

    printf("CLIENTE C\n");
    //Menu
    while(option != 3){

        Sock = socket (AF_INET, SOCK_STREAM, 0);

        printMenu(&option);

        switch (option)
        {
        case 1:
            //INGRESO LONGITUD
            printf("Ingrese la longitud del nombre de usuario: ");
            readLength(&length);

            //ENVIAR REQUEST
            sendData(option, length,Sock, &serverAddress);

            //RECIBIR RESPUESTA
            recieve(Sock,5,15);

            //CERRAR CONEXION
            closesocket(Sock);
            break;
        case 2:
            //INGRESO LONGITUD
            printf("Ingrese la longitud de la contrasenia: ");
            readLength(&length);

            //ENVIAR REQUEST
            sendData(option, length,Sock, &serverAddress);

            //RECIBIR RESPUESTA
            recieve(Sock,8,49);

            //CERRAR CONEXION
            closesocket(Sock);
            break;
        case 3:
            printf("Saliendo...\n");
             /*Cerrar conexion*/
            closesocket(Sock);
            WSACleanup();

            Sleep(1000);
            return;
            break;
        default:
            printf("\033[1;31mDebe ingresar un valor del 1 al 3\033[0m\n");
            break;
        }

    }
}